compiler/damlc/tests/daml-test-files/TemplateLetWithSignatures.daml

```diff
@@ -0,0 +1,28 @@
+-- Copyright (c) 2020, Digital Asset (Switzerland) GmbH and/or its affiliates.
+-- All rights reserved.
+
+module TemplateLetWithSignatures where
+
+-- This example should fail to type check.
+-- There was a bug reported: https://github.com/digital-asset/daml/issues/7902
+-- which indicated that the signature was being ignored.
+-- It is now not ignored, and the example does indeed fail to type check.
+
+-- TODO: only report the type error once, instead of 5 times!
```

Comment:
The more choices you add, the more errors you get.

Do you have an idea how to fix this?

Reply:
I guess a new top level function which is passed this and returns a tuple of the values, which get bound to the vars in each place.

Reply:
This should get fixed as when issue  https://github.com/digital-asset/daml/issues/2175 is addressed.